internal/sync-pnpm/index.js

```diff
@@ -0,0 +1,41 @@
+import { join, dirname } from 'node:path';
+import { createRequire } from 'node:module';
+
+import { getPackages } from '@manypkg/get-packages';
+import { findRoot } from '@manypkg/find-root';
+import { readJson, pathExists } from 'fs-extra/esm';
+import { hardLinkDir } from '@pnpm/fs.hard-link-dir';
```

Comment:
Ahhhh - this is interesting!

> Hard link (or copy if linking fails) all files from a directory to several target directories.

So essentially this sync file forces `pnpm` to hard link the dependencies one way or another so that we don't have to continually `pnpm i -f` anytime we make a change in the addon source and want to see it in the test-app/docs-app?

Reply:
Yes, exactly. Also it does _only_ this, it does not try to re-install/shuffle things around. 

In an earlier attempt, I tried to just use `pnpm i -f` in a turbo step, but when things run in parallel, this would lead to weird errors, as `pnpm i -f` would temporarily remove things from `node_modules` and add them back moments later, but in the meantime some parallel build would just fail as it would miss some dependencies in `node_modules`.

So this does just adds back the missing hard links of our `/dist` folder, and nothing else.

Reply:
Amazing!!

So is the current development process now (for something like `docs-app`):
1. make a change in `ember-headless-form`
2. `pnpm build`
3. see the update in the `docs-app`?

Reply:
Yes, exactly, that should work (assuming you have `pnpm start` or `pnpm start:docs` running, to continuously rebuild the app).

It is a bit of a regression compared to other v2 addon setups (e.g. using `yarn`), where this `injected` thing does not exist (but instead other peer dependency related problems), so you could just run `start` and it would watch the addon and the test-app. But not a regression compared to the previous state of this addon, because we had to run `pnpm i -f` again and again, so this should make it better at least!

Reply:
Just realized that the `pnpm build` script does too much for this use case, as it builds the addons _and_ the apps (test and docs). Will need another iteration!